core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -90,8 +90,13 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
-    if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDirectory) {
-      throw new IllegalArgumentException(s"Log directory $logBaseDir does not exist.")
+    try {
+      if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDirectory) {
+        throw new IllegalArgumentException(s"Log directory $logBaseDir is not a directory.")
+      }
+    } catch {
+      case f: FileNotFoundException =>
```

Comment:
In several cases above it might be fine to change to let the FNFE fly rather than change it to an IAE with another message. Not out of the question. This looks good though.